src/client/mod.rs

```diff
@@ -825,15 +850,66 @@ impl KafkaClient {
                                                                           now,
                                                                           req));
             for tp in resp.topic_partitions {
-                let e = res.entry(tp.topic).or_insert(vec![]);
-                for p in tp.partitions {
-                    e.push(p.into_offset());
+                let mut entry = res.entry(tp.topic);
+                let mut new_resp_offsets = None;
+                let mut err = None;
+
+                // Use an explicit scope here to allow insertion into a vacant entry
+                // below
```

Comment:
no need for this comment. the borrow checker will hit us immediately when we try not to use the extra scope :)